test/e2e/src/test-contract-client-constructor.js

```diff
@@ -0,0 +1,110 @@
+const test = require('ava')
+const { spawnSync } = require('node:child_process')
+const { Address } = require('../../..')
+const { contracts, networkPassphrase, rpcUrl, friendbotUrl } = require('./util')
+const { ContractSpec } = require('../../..')
+const { Keypair } = require('../../..')
+
+const {
+  ContractClient,
+  basicNodeSigner,
+} = require('../../../lib/contract_client')
```

Comment:
If you note the tests in the non-e2e directories, they actually pull from `StellarSdk` just like a downstream app would, where StellarSdk gets injected into `global`/`window` (see `test-nodejs.js` and` test-browser.js`). This directory uses a different framework, but we similarly want to have both browser and Node tests, but I'm not sure how to accomplish that since I didn't write these tests (cc @chadoh, would you know?).

Maybe this is a reflection of why we didn't catch the `ScVal` object thing if the other tests are running in the browser, either?

Reply:
I didn't know the other tests were running in node and browser! Very cool. No, these e2e tests only run in node. Would be very cool to have them run in the browser, too. I [noted in the original ContractClient PR](https://github.com/stellar/js-stellar-sdk/pull/929) that the tests were technical debt. We should probably make an issue to track this, and make sure that we also make them work in the browser.

For now, this test is following the convention of all the other e2e tests, so maybe we leave as-is and then fix it up to match #962 once that gets merged, and finally fix them all up to run in the browser too (and maybe use mocha, like the other tests, but TBD?) as part of that follow-up.

Reply:
I can leave the tests as is for now as Chad suggest and work on the e2e tests later on (its on my todo list). I think we just need to decide on if we want to convert them all to mocha or just keep using ava and I guess add browser versions of the test?

Reply:
Agreed, let's punt it :+1: imo browser versions >>> same framework but I guess both would be ideal